app/src/main/java/com/android/sample/model/hour_date/HourDateViewModel.kt

```diff
@@ -49,17 +50,26 @@ open class HourDateViewModel : ViewModel() {
 
   fun combineDateAndTime(date: Timestamp, time: String): Timestamp {
     // Convert Firebase Timestamp to LocalDate
-    val localDate = date.toDate().toInstant().atZone(ZoneId.systemDefault()).toLocalDate()
+    try {
+      // Convert Firebase Timestamp to LocalDate
+      val localDate = date.toDate().toInstant().atZone(ZoneId.systemDefault()).toLocalDate()
 
-    val formatter = DateTimeFormatter.ofPattern("HH:mm")
-
-    val localTime = LocalTime.parse(time, formatter) // Assuming "hh:mm" format
+      val formatter = DateTimeFormatter.ofPattern("HH:mm")
+      val localTime = LocalTime.parse(time, formatter) // Parse the time using the provided format
 
-    // Combine LocalDate and LocalTime to LocalDateTime
-    val combinedDateTime = LocalDateTime.of(localDate, localTime)
+      // Combine LocalDate and LocalTime to LocalDateTime
+      val combinedDateTime = LocalDateTime.of(localDate, localTime)
 
-    // Convert back to Firebase Timestamp
-    return Timestamp(
-        combinedDateTime.atZone(ZoneId.systemDefault()).toInstant().toEpochMilli() / 1000, 0)
+      // Convert back to Firebase Timestamp
+      return Timestamp(
+          combinedDateTime.atZone(ZoneId.systemDefault()).toInstant().toEpochMilli() / 1000, 0)
+    } catch (e: DateTimeParseException) {
+      // Log the error or handle it as needed
```

Comment:
This is an acceptable solution as It does not crash the app but if we're getting the activity's TimeStamp we cannot give it the current time if It fails. We can instead throw exceptions if the parameters of the method " fun combineDateAndTime(date: Timestamp, time: String): Timestamp {" do not provide the correct parameters.
I have seen in that the app crash at: com.android.sample.ui.listActivities.ViewListActivitiesKt$ListActivitiesScreen$4.invoke(ViewListActivities.kt:178)
maybe we could instead check that the provided date is a Timestamp is not null and if the time is in the correct format: "HH:MM".